frontend/src/components/ShootWorkers/MachineType.vue

```diff
@@ -71,62 +63,110 @@ export default {
     HintColorizer
   },
   props: {
-    worker: {
-      type: Object,
+    value: {
+      type: String,
       required: true
     },
+    searchInput: {
+      type: String
+    },
+    valid: {
+      type: Boolean
+    },
     machineTypes: {
       type: Array,
       default: () => []
     }
   },
   data () {
     return {
-      validationErrors,
-      valid: undefined
+      lazyValue: this.value,
+      lazySearch: this.searchInput,
+      validationErrors
     }
   },
   computed: {
+    internalValue: {
+      get () {
+        return this.lazyValue
+      },
+      set (value) {
+        this.lazyValue = value ?? ''
+        this.$emit('input', this.lazyValue)
+      }
+    },
+    internalSearch: {
+      get () {
+        return this.lazySearch
+      },
+      set (value) {
+        if (this.lazySearch !== value) {
+          this.lazySearch = value
+          this.$emit('update:search-input', value)
+        }
+      }
+    },
     machineTypeItems () {
       const machineTypes = [...this.machineTypes]
-      if (this.notInList) {
+      if (this.notInList && this.internalValue) {
         machineTypes.push({
-          name: this.worker.machine.type
+          name: this.internalValue
         })
       }
       return machineTypes
     },
     notInList () {
       // notInList: selected value may have been removed from cloud profile or other worker changes do not support current selection anymore
-      return !find(this.machineTypes, ['name', this.worker.machine.type])
+      return !find(this.machineTypes, ['name', this.internalValue])
     },
     hint () {
-      if (this.notInList) {
-        return 'This machine type may not be supported by your worker'
-      }
-      return undefined
+      return this.notInList ? 'This machine type may not be supported by your worker' : ''
     }
   },
   validations,
   methods: {
     getErrorMessages (field) {
       return getValidationErrors(this, field)
     },
-    onInputMachineType () {
-      this.$v.worker.machine.type.$touch()
-      this.$emit('update-machine-type', this.worker.machine.type)
-      this.validateInput()
-    },
-    validateInput () {
-      if (this.valid !== !this.$v.$invalid) {
-        this.valid = !this.$v.$invalid
-        this.$emit('valid', { id: this.worker.id, valid: this.valid })
+    filter (item, query) {
+      if (!item) {
+        return false
+      }
+      if (typeof query !== 'string') {
+        return true
+      }
+      const { name, cpu, gpu, memory, storage } = item
+      const terms = query.split(/\s+/)
+      const properties = [cpu, gpu, memory]
+      if (storage) {
+        properties.push(storage.type)
+        properties.push(storage.class)
+        properties.push(storage.size)
       }
+      return terms.every(term => name?.includes(term) || properties.includes(term))
```

Comment:
```suggestion
      return terms.every(term => name?.includes(term) || some(properties, prop => prop?.includes(term)))
```
I would change it to something like this to allow partial match also for the properties. Moreover, you could do a `toLowerCase()` for all the strings before comparing them.

Reply:
✔️